rust_src/src/numbers.rs

```diff
@@ -39,3 +47,40 @@ fn numberp(object: LispObject) -> LispObject {
 fn number_or_marker_p(object: LispObject) -> LispObject {
     LispObject::from_bool(object.is_number() || object.is_marker())
 }
+
+/// Return a pseudo-random number.
+/// All integers representable in Lisp, i.e. between `most-negative-fixnum'
+/// and `most-positive-fixnum', inclusive, are equally likely.
+///
+/// With positive integer LIMIT, return random number in interval [0,LIMIT).
+/// With argument t, set the random number seed from the system's entropy
+/// pool if available, otherwise from less-random volatile data such as the time.
+/// With a string argument, set the seed based on the string's contents.
+/// Other values of LIMIT are ignored.
+///
+/// See Info node `(elisp)Random Numbers' for more details.
+#[lisp_fn(min = "0")]
+fn random(limit: LispObject) -> LispObject {
+    let mut rng = RNG.lock().unwrap();
```

Comment:
I presume that this will change the output of the random number generator in Remacs, even when seeded with the same value as GNU Emacs? If so, we should mention it in REMACS_COMPATIBILITY.md